src/resolver.ts

```diff
@@ -124,11 +145,11 @@ export function inMemoryCache(): DIDCache {
 
     const cached = cache.get(parsed.didUrl)
     if (cached !== undefined) return cached
-    const doc = await resolve()
-    if (doc !== null) {
-      cache.set(parsed.didUrl, doc)
+    const result = await resolve()
+    if (result.didResolutionMetadata?.error !== 'notFound') {
```

Comment:
Why do you think errors should be cached?

Reply:
I just don't see why `notFound` should be treated differently from when something is found. Doesn't matter too much though.

Reply:
no, I mean why cache ANY error results?